program/src/processor.rs

```diff
@@ -435,74 +444,93 @@ fn process_switch_vote(
         let state = bytemuck::try_from_bytes_mut::<ProposalVote>(&mut data)
             .map_err(|_| ProgramError::InvalidAccountData)?;
 
-        let pod_new_vote = PodBool::from(new_vote);
-        if state.vote == pod_new_vote {
+        if state.election == new_election {
```

Comment:
There's still an edge case -- if I just want to update the stake weight on my vote (ie. I'm still `For`, but I now have 110 stake instead of 100), I need to switch it to something else, and then switch it again to `For`.

Since the later logic handles all cases, how about we only short-circuit `if state.election == election && state.stake == stake`?

Reply:
I'll work in some test updates as well, but I think that's all that's needed in the processor code.

Reply:
Works for me!